platform/api.scripting/src/org/netbeans/modules/scripting/GraalEngine.java

```diff
@@ -0,0 +1,175 @@
+        return unbox(result);
+    }
+
+    private Value evalImpl(ScriptContext arg1, String src) throws ScriptException {
+
+        try {
+            return ((GraalContext) arg1).ctx().eval(id(), src);
+        } catch (PolyglotException e) {
+            throw new ScriptException(e);
+        }
+    }
+
+    @Override
+    public Object eval(Reader arg0, ScriptContext arg1) throws ScriptException {
+        Source src = Source.newBuilder(id(), arg0, null).buildLiteral();
+        Value result = ((GraalContext)arg1).ctx().eval(src);
+        return unbox(result);
+    }
+
+    @Override
+    public Object eval(String arg0) throws ScriptException {
+        return eval(arg0, factory.ctx);
+    }
+
+    @Override
+    public Object eval(Reader arg0) throws ScriptException {
+        return eval(arg0, factory.ctx);
+    }
+
+    @Override
+    public Object eval(String arg0, Bindings arg1) throws ScriptException {
+        return eval(arg0);
+    }
+
+    @Override
+    public Object eval(Reader arg0, Bindings arg1) throws ScriptException {
+        return eval(arg0);
+    }
+
+    @Override
+    public void put(String arg0, Object arg1) {
+    }
+
+    @Override
+    public Object get(String arg0) {
+        return null;
+    }
+
+    @Override
+    public Bindings getBindings(int scope) {
+        return getContext().getBindings(scope);
+    }
+
+    @Override
+    public void setBindings(Bindings arg0, int arg1) {
+    }
+
+    @Override
+    public Bindings createBindings() {
+        return null;
+    }
+
+    @Override
+    public void setContext(ScriptContext arg0) {
+        throw new IllegalStateException();
+    }
+
+    @Override
+    public Object invokeMethod(Object thiz, String name, Object... args) throws ScriptException, NoSuchMethodException {
+        final Value thisValue = (Value) thiz;
+        Value fn = thisValue.getMember(name);
+        Value result = fn.execute(args);
+        return unbox(result);
+    }
+
+    @Override
+    public Object invokeFunction(String name, Object... args) throws ScriptException, NoSuchMethodException {
+        final Value fn = evalImpl(getContext(), name);
+        final Value result = fn.execute(args);
+        return unbox(result);
+    }
+
+    @Override
+    public <T> T getInterface(Class<T> clasz) {
+        try {
+            return getInterface(evalImpl(getContext(), "this"), clasz);
+        } catch (ScriptException ex) {
+            throw raise(RuntimeException.class, ex);
+        }
+    }
+
+    @Override
+    public <T> T getInterface(Object thiz, Class<T> clasz) {
+        if (thiz instanceof Value) {
+            return ((Value) thiz).as(clasz);
+        }
+        if (clasz.isInstance(thiz)) {
+            return clasz.cast(thiz);
+        }
+        return null;
+    }
+
+    private Object unbox(Value result) {
+        if (result.isNull()) {
+            return null;
+        }
+        if (result.isNumber()) {
+            return result.as(Number.class);
```

Comment:
Isn't further classification using `fitsInByte` etc needed ?

Reply:
Is it? I don't know. Any example you have in mind?